executor/builder.go

```diff
@@ -1556,6 +1556,61 @@ func (b *executorBuilder) buildIndexLookUpJoin(v *plannercore.PhysicalIndexJoin)
 	e.innerCtx.keyCols = innerKeyCols
 	e.joinResult = e.newFirstChunk()
 	metrics.ExecutorCounter.WithLabelValues("IndexLookUpJoin").Inc()
+
+	// If the inner join keys are the suffix set of the index, then return the IndexLookUpMergeJoin
+	var isIndexScan bool
+	var is *plannercore.PhysicalIndexScan
+	if ir, ok := innerPlan.(*plannercore.PhysicalIndexReader); ok {
+		is, isIndexScan = ir.IndexPlans[0].(*plannercore.PhysicalIndexScan)
+	}
+	if ilr, ok := innerPlan.(*plannercore.PhysicalIndexLookUpReader); ok {
+		is, isIndexScan = ilr.IndexPlans[0].(*plannercore.PhysicalIndexScan)
+	}
+	if isIndexScan {
+		idx := is.Index.Columns
+		if len(v.InnerJoinKeys) <= len(idx) {
```

Comment:
If `InnerJoinKeys` are not the prefix of `idx`, we can't guarantee the inner results ordered by `InnerJoinKeys`. Then it will cause `MergeJoin` incorrect.

Reply:
So can you give an example when it is false? Additionally, for index (b, c), if  
the join key is `c` and there is a condition of `b = 1`, I think we can still promise the order.

Reply:
You are right. All the other none-index join keys are transferred to other conditions.